agent.go

```diff
@@ -224,6 +255,19 @@ func (c *container) getProcess(execID string) (*process, error) {
 	return proc, nil
 }
 
+func (s *sandbox) trace(name string) (opentracing.Span, context.Context) {
+	if s.ctx == nil {
+		agentLog.WithField("type", "bug").Error("trace called before context set")
```

Comment:
might be this function should be a `singleton`, hence log an error is not needed

Reply:
I added this for consistency with the approach used in the runtime. If you look at `agent.go`, you'll see that there is a "gap" between creating the sandbox and having `s.ctx` set. That is unfortunately required since:

- the context can only be set after tracing has been setup.
- the sandbox needs to be created before tracing has been setup.

We could use a new context when creating the sandbox and then reset it to the correct context containing the trace metadata but this approach is simpler I think.